test/test-apimatic-core/union_type_tests/one_of_test.rb

```diff
@@ -0,0 +1,1745 @@
+require 'minitest/autorun'
+require 'apimatic_core'
+
+require_relative '../../test-helper/models/morning'
+require_relative '../../test-helper/models/evening'
+require_relative '../../test-helper/models/month_number_enum'
+require_relative '../../test-helper/models/month_name_enum'
+
+class OneOfTest < Minitest::Test
+  include CoreLibrary, TestComponent
+
+  def setup
+  end
+
+  def teardown
+  end
+
+  # === Simple cases ===
+
+  def test_valid_string_in_one_of
+    _one_of = OneOf.new([LeafType.new(String), LeafType.new(Integer)])
+    _one_of.validate('string')
+    assert _one_of.is_valid
+  end
+
+  def test_valid_integer_in_one_of
+    _one_of = OneOf.new([LeafType.new(String), LeafType.new(Integer)])
+    _one_of.validate(4)
+    assert _one_of.is_valid
+  end
+
+  def test_valid_float_in_one_of
+    _one_of = OneOf.new([LeafType.new(String), LeafType.new(Float)])
+    _one_of.validate(4.0)
+    assert _one_of.is_valid
+  end
+
+  def test_valid_integer_float_in_one_of
+    _one_of = OneOf.new([LeafType.new(Integer), LeafType.new(Float)])
+    _one_of.validate(4)
+    assert _one_of.is_valid
+  end
+
+  def test_valid_boolean_class_float_in_one_of
+    _one_of = OneOf.new([LeafType.new(Integer), LeafType.new(TrueClass), LeafType.new(FalseClass)])
+    _one_of.validate(false)
+    assert _one_of.is_valid
+  end
+
+  def test_valid_mutiple_types_in_one_of
+    _one_of = OneOf.new([LeafType.new(Integer), LeafType.new(Float), LeafType.new(String)])
+    _one_of.validate(4)
+    assert _one_of.is_valid
+  end
+
+  def test_valid_both_string_one_of
+    _one_of = OneOf.new([LeafType.new(String), LeafType.new(String)])
+    assert_raises OneOfValidationException do
```

Comment:
Blocker: The assertion against the error messages that the exception has thrown is missing. I'd recommend to see the tests in python core library.

Reply:
done